src/cursor/cur_std.c

```diff
@@ -1113,6 +1113,51 @@ __wt_cursor_reconfigure(WT_CURSOR *cursor, const char *config)
     API_END_RET(session, ret);
 }
 
+/*
+ * __wt_cursor_largest_key --
+ *     WT_CURSOR->largest_key default implementation..
+ */
+int
+__wt_cursor_largest_key(WT_CURSOR *cursor)
+{
+    WT_DECL_ITEM(key);
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    WT_TXN *txn;
+
+    txn = NULL;
+    CURSOR_API_CALL(cursor, session, largest_key, NULL);
+
+    txn = session->txn;
+    if (F_ISSET(txn, WT_TXN_SHARED_TS_READ))
+        WT_ERR_MSG(session, EINVAL, "largest key cannot be called with a read timestamp");
+
+    WT_ERR(__wt_scr_alloc(session, 0, &key));
+
+    /* Reset the cursor to give up the cursor position. */
+    WT_ERR(cursor->reset(cursor));
+    /* Ignore deletion */
+    F_SET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);
+
+    /* Call cursor prev with read uncommitted isolation level. */
+    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED, ret = cursor->prev(cursor));
+    WT_ERR(ret);
+
+    /* Copy the key as we will reset the cursor after that. */
+    WT_ERR(__wt_buf_set(session, key, cursor->key.data, cursor->key.size));
+    WT_ERR(cursor->reset(cursor));
```

Comment:
Can you explain why we need to reset the cursor here?

Reply:
We want the cursor not positioned after the call so we reset the cursor here.